lib/puppet/configurer.rb

```diff
@@ -157,7 +157,10 @@ def prepare_and_retrieve_catalog(options, query_options)
 
     # apply passes in ral catalog
     catalog = options.delete(:catalog)
-    return catalog if catalog
+    if catalog
+      @cached_catalog_status = 'explicitly_requested_via_apply'
```

Comment:
Is this misleading? This value ends up in the report under `cached_catalog_status`, which makes it seem as if one was used here, even though a catalog was supplied instead. Since this case is specific to `apply`, my main question is whether reports are generated from applies, and if so, if it seems wrong to report this here instead of `not_used`.

Reply:
Agreed. It IS added to the report.  So, I'll find another way to do this.